Fill.py

```diff
@@ -15,8 +17,71 @@ class FillError(RuntimeError):
     pass
 
 
-def _log_fill_progress(name: str, placed: int, total_items: int) -> None:
-    logging.info(f"Current fill step ({name}) at {placed}/{total_items} items placed.")
+class FillLogger():
+    min_size: int = 1000   # minimum number of items we care about reporting (non-tty)
+    min_time: float = 0.25 # minimum time (in seconds) between reports (tty)
+
+    def __init__(self, total_items: int):
+        self.start_time = time.time()
+        self.prev_time = time.time()
+        self.cur_time = time.time()
+        self.step: int = max(round(total_items * 0.1), 1000)
+        self.total_items: int = total_items
+        self.is_debug: bool   = False
```

Comment:
this one is still overindented

Reply:
arg, sorry; i forgot to check PEP8 with the debug update. Good catch. I'll do a more thorough pass as well as fix this specific.